phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/db/cloud/OutgoingType.kt

```diff
@@ -20,15 +19,17 @@ data class OutgoingPaymentWrapper @OptIn(ExperimentalSerializationApi::class) co
     val recipient: ByteArray,
     val details: DetailsWrapper,
     val parts: List<OutgoingPartWrapper>,
+    val closingTxsParts: List<OutgoingClosingTxPartWrapper> = emptyList(),
```

Comment:
@robbiehanson I would like to have your input on this change.

We have several options to store closing tx parts in this wrapper while ensuring backward compatibility:

**Option 1** : `parts` becomes `part: List<PartWrapper>`, where:
```kotlin
@SerialName("lightning")
data class LightningPart(...): PartWrapper
// json => { "type": "lightning", "id": 123... }

@SerialName("closingtx")
data class ClosingTxWrapper(...): PartWrapper
// json => { "type": "closingtx", "id": 123... }
```
It looks great but causes backward compatibility issues. What to do with legacy data which don't have typed parts? The deserializer does not handle that well.

Solutions could then be :
- add a version to `OutgoingPaymentWrapper` (like `CloudData` is) and do some special rules for data without a version (i.e. those data contain untyped parts);
- or find some trick to let the serializer handle the "untyped" legacy data parts as `LightningPart`s by default.


**Option 2** : we store lightning parts and closing txs parts separately. It is backward compatible, does not add much code, but it is quite ugly.

So I've chosen option 2 for now, because it's the easiest but it may not be the best long term. Wdyt?

Reply:
Thanks! I'll keep it that way then.